samples/rpc/pytest/test_sample.py

```diff
@@ -0,0 +1,65 @@
+# Copyright (c) 2022 Golioth, Inc.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import os
+import sys
+from pathlib import Path
+from datetime import datetime, timedelta
+import logging
+
+from golioth import Client, RPCTimeout
+import pytest
+from trio import sleep
+
+
+# This is the same as using the @pytest.mark.anyio on all test functions in the module
+pytestmark = pytest.mark.anyio
+
+
+DEFAULT_TIMEOUT = 30
+
+
+@pytest.fixture(scope='session')
+def initial_timeout(request):
+    timeout = request.config.getoption('--initial-timeout')
+    if timeout is None:
+        timeout = DEFAULT_TIMEOUT
+    return timeout
+
+
+@pytest.fixture(scope='session')
+async def device(initial_timeout):
+    device_name = os.environ["GOLIOTH_DEVICE_NAME"]
+
+    client = Client(os.environ.get("GOLIOTHCTL_CONFIG"))
+    project = await client.default_project()
+    device = await project.device_by_name(device_name)
+
+    # Try to call method for the first time, just to make sure that device is already available
+    # (already connected).
+    start = datetime.now()
+
+    while datetime.now() < start + timedelta(seconds=initial_timeout):
+        try:
+            logging.debug('Calling "multiply" to check if device is connected')
+            await device.rpc.multiply(1, 1)
+
+            # No exception was raised, so just return device
+            return device
+        except RPCTimeout:
```

Comment:
One thing I've been trying to capture lately (without success) is the latency of RPCs (min, average, max, etc). I suspect there's some kind of issue on the backend, since I sometimes see 2-3 second response times, but I haven't been able to prove it. I think this test (or something like it), could be used to call the RPC 1000 times and measure the results. I'm looking forward to it!

Reply:
Definitely. There is also `pytest-repeat` plug, which allows you to to something similar without even adjusting this test. There are other plugins for generating reports as well, which should also include such timing information.